pkg/controllers/provisioning/scheduling/nodeclaim.go

```diff
@@ -251,10 +252,22 @@ func filterInstanceTypesByRequirements(instanceTypes []*cloudprovider.InstanceTy
 		fitsAndOffering:         false,
 	}
 
-	for _, it := range instanceTypes {
+	for _, it := range n.InstanceTypeOptions {
```

Comment:
I wonder if we should collapse this code with https://github.com/kubernetes-sigs/karpenter/pull/1379/files, which aims to solve this type of problem

Reply:
I think this might lead to referring instanceType provider from more packages than it really needs to be referred from...